src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -4774,6 +4862,12 @@ private bool PathIsReservedDeviceName(string destinationPath, string errorId)
             return pathIsReservedDeviceName;
         }
 
+        private long _totalFiles = 0;
+        private long _totalBytes = 0;
+        private long _copiedFiles = 0;
+        private long _copiedBytes = 0;
+        private DateTime _copyStartTime;
```

Comment:
Design comments say it is preferable to have FileSystemProvider _stateless_.
Really all the fields can be local variables (AsyncLocal).

Reply:
Having changed from a Thread to a Task, using AsyncLocal for these variables didn't work as expected.  This is type is new to me, so perhaps you can figure out how to make it work and add it to this PR?